exercises/yacht/examples/success-standard/src/Yacht.hs

```diff
@@ -0,0 +1,59 @@
+module Yacht (yacht, Category(..)) where
+
+import Control.Monad (join)
+import Data.List (sortOn, sort, group)
+
+data Category = Ones
+              | Twos
+              | Threes
+              | Fours
+              | Fives
+              | Sixes
+              | FullHouse
+              | FourOfAKind
+              | LittleStraight
+              | BigStraight
+              | Choice
+              | Yacht
+
+yacht :: Category -> [Int] -> Int
+yacht Ones   = valueScore 1
+yacht Twos   = valueScore 2
+yacht Threes = valueScore 3
+yacht Fours  = valueScore 4
+yacht Fives  = valueScore 5
+yacht Sixes  = valueScore 6
+
+yacht FullHouse = sum
+                . join
+                . (\xs -> if length xs == 2 then xs else [])
+                . filter ((<= 3) . length)
+                . diceGroups
+
+yacht FourOfAKind = sum
+                  . take 4
+                  . join
+                  . filter ((>= 4) . length)
+                  . diceGroups
+
+yacht LittleStraight = verify . sort
+  where verify [1, 2, 3, 4, 5] = 30
+        verify _ = 0
+
+yacht BigStraight = verify . sort
+  where verify [2, 3, 4, 5, 6] = 30
+        verify _ = 0
+
+yacht Choice = sum
+
+yacht Yacht = (* 10)
+            . length
+            . join
+            . filter((== 5) . length)
+            . diceGroups
```

Comment:
I could make a `filterJoinGroups` function to consolidate what happens for these last 3 lines and for what happens in `FourOfAKind`, but I didn't know if that was warranted.